functional-programming/FP.md

```diff
@@ -16,26 +19,42 @@
 - mutation
 - map
 - filter
+- reduce
 
 ### Materials
 
 -[What is Functional Programming[(https://medium.com/javascript-scene/master-the-javascript-interview-what-is-functional-programming-7f218c68b3a0)
+-[Map, Filter, and Reduce in JavaScript (https://medium.com/jsguru/javascript-functional-programming-map-filter-and-reduce-846ff9ba492d)]
 - <img width="608" alt="steven_luscher_on_twitter___map_filter_reduce_in_a_tweet__map__ __ __ ___cook______ __ __ __filter__ __ __ ___isvegetarian______ __ __reduce__ __ ___eat_____ _" src="https://cloud.githubusercontent.com/assets/16581819/22812830/e2d723fe-eefb-11e6-87ed-775e2b5fd105.png">
 - [This other website](otherexample.com)
 
 ### Lesson
+Here are a list of tools that become available with functional programming. They are common functions that take apply functions across all elements in a container, such as an array.
 
-    var array = [1, 2, 3]
+Part I. Map
+The purpose of `map` is to process every element of an array in the exact same way. When you call `map` on an array, it executes that callback on every element within it, returning a new array with all of the values that the callback returned.
+Example: array of integers, add '3' to every integer
+    var intArray = [1, 2, 3]
 
     function add3(value) {
       return value + 3;
     }
 
-    array.map(add3)
+    array.intArray(add3)
```

Comment:
Can you show them the output too?  Also, I'm afraid I don't know what line 43 means, and are you still using map?